### PR TITLE
docs(python): map-related docstring updates

### PR DESCRIPTION
**Pull request by alexander-beedie**
Various updates for assorted `map_*` docstrings. Includes additional information, minor clarifications, and a new example showing window function context via `over` grouping.

### CODE REVIEW

File: py-polars/polars/lazyframe/frame.py [hunk 1]
```diff
@@ -5172,20 +5172,27 @@ def map_batches(
         --------
         >>> lf = pl.LazyFrame(
         ...     {
-        ...         "a": [1, 2],
-        ...         "b": [3, 4],
+        ...         "a": range(-100_000, 0),
+        ...         "b": range(0, 100_000),
```
Comment (stinodego) [inline py-polars/polars/lazyframe/frame.py:5176]: Shall we use `int_range` here?

```suggestion
        ...         "a": pl.int_range(-100_000, 0),
        ...         "b": pl.int_range(0, 100_000),
```

File: py-polars/polars/lazyframe/frame.py [hunk 2]
```diff
@@ -5172,20 +5172,27 @@ def map_batches(
         --------
         >>> lf = pl.LazyFrame(
         ...     {
-        ...         "a": [1, 2],
-        ...         "b": [3, 4],
+        ...         "a": range(-100_000, 0),
```
Comment (stinodego) [inline py-polars/polars/lazyframe/frame.py:5175]: I don't really like that we have such an expensive test in our doctests. I get the idea of showing a more realistic example, but maybe a range of `1000` elements is enough? Or keep 100_000 and just skip this example?

**Review by stinodego — COMMENTED** [review]
Thanks! Two minor comments.

File: py-polars/polars/lazyframe/frame.py (hunk 2 above)
Comment (alexander-beedie) [reply py-polars/polars/lazyframe/frame.py:5175]: Good idea - adding the SKIP makes a lot of sense there; some of our examples are so trivial they're not really showing anything, so it's good to have something that feels at least a _little_ more "real" 😅

File: py-polars/polars/lazyframe/frame.py (hunk 1 above)
Comment (alexander-beedie) [reply py-polars/polars/lazyframe/frame.py:5176]: Sure, can do - we do actually recognise/specialise python `range` internally though (for a huge speedup, of course :), so there's no performance difference, but it's a good idea to draw attention to the native variant 👍

File: py-polars/polars/lazyframe/frame.py (hunk 2 above)
Comment (alexander-beedie) [reply py-polars/polars/lazyframe/frame.py]: Done 👍

File: py-polars/polars/lazyframe/frame.py (hunk 1 above)
Comment (stinodego) [reply py-polars/polars/lazyframe/frame.py]: > so there's no performance difference

Ah, I didn't know that. If I knew that, I would've left this. But either is fine I guess.

**Review by stinodego — APPROVED** [review]
:tada:

File: py-polars/polars/lazyframe/frame.py (hunk 1 above)
Comment (alexander-beedie) [reply py-polars/polars/lazyframe/frame.py]: > Ah, I didn't know that. If I knew that, I would've left this. But either is fine I guess.

I only know about it as it was one of mine...🤣 
https://github.com/pola-rs/polars/pull/5397

**Pull request merged**